src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/FileBinding.java

```diff
@@ -48,13 +43,7 @@
         return FileCredentials.class;
     }
 
-    @Override public SingleEnvironment bindSingle(Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-        FileCredentials credentials = getCredentials(build);
-        FilePath secrets = secretsDir(workspace);
-        String dirName = UUID.randomUUID().toString();
-        final FilePath dir = secrets.child(dirName);
-        dir.mkdirs();
-        secrets.chmod(/*0700*/448);
+    @Override protected final FilePath write(FileCredentials credentials, FilePath dir) throws IOException, InterruptedException {
         FilePath secret = dir.child(credentials.getFileName());
         copy(secret, credentials);
         if (secret.isDirectory()) { /* ZipFileBinding */
```

Comment:
Yes, you're probably right. I was only keeping this code around to avoid breaking other (hypothetical) sub-classes of FileBinding (for instances implementations for different types of archives), but a Github search for "FileBinding" on jenkinsci finds no such thing, so I guess it's not a real concern.

I will add a 2nd commit to the PR.